x-pack/legacy/plugins/reporting/server/routes/lib/make_request_facade.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RequestQuery } from 'hapi';
+import { Legacy } from 'kibana';
+import {
+  RequestFacade,
+  ReportingRequestPayload,
+  ReportingRequestPre,
+  ReportingRequestQuery,
+} from '../../../types';
+
+export function makeRequestFacade(request: Legacy.Request): RequestFacade {
+  // This condition is for unit tests
+  const getSavedObjectsClient = request.getSavedObjectsClient
+    ? request.getSavedObjectsClient.bind(request)
+    : request.getSavedObjectsClient;
+  return {
+    getSavedObjectsClient,
+    headers: request.headers,
+    auth: request.auth, // for getUser
+    params: request.params,
+    payload: (request.payload as object) as ReportingRequestPayload,
+    query: ((request.query as RequestQuery) as object) as ReportingRequestQuery,
+    pre: (request.pre as Record<string, any>) as ReportingRequestPre,
+    getBasePath: request.getBasePath,
+    route: request.route,
+    raw: request.raw,
+  };
```

Comment:
Following @restrry comment in https://github.com/elastic/kibana/pull/54082/files#r365183259: maybe you should return an enhanced request here instead of picking properties. This way we would ensure that external calls relying on an actual request object (such as `server.plugins.security.getUser`) would not break, while preserving your type. This is not the most graceful thing to do in TS for sure, but it may be 'safer' in real time situation. WDTY both ?

```typescript
export function makeRequestFacade(request: Legacy.Request): RequestFacade {
  // This condition is for unit tests
  const getSavedObjectsClient = request.getSavedObjectsClient
    ? request.getSavedObjectsClient.bind(request)
    : request.getSavedObjectsClient;
  return {
    ...request,
    getSavedObjectsClient,
  } (as Unknown probably) as RequestFacade;
```

Reply:
Following that comment, I liked the advice given by @restrry [here](https://github.com/elastic/kibana/pull/54082#discussion_r365679676) so I added 80e0fcc

Reply:
Did not saw his last comment. This is way better than my suggestion.

Reply:
Oops, I mean cb40fa3896

See https://github.com/elastic/kibana/pull/54082#discussion_r366589274